avm/res/compute/virtual-machine/main.bicep

```diff
@@ -618,19 +813,125 @@ module vm_azureDiskEncryptionExtension 'extension/main.bicep' = if (extensionAzu
+  params: {
+    virtualMachineName: vm.name
+    name: 'HostPoolRegistration'
+    location: location
+    publisher: 'Microsoft.PowerShell'
+    type: 'DSC'
+    typeHandlerVersion: contains(extensionHostPoolRegistration, 'typeHandlerVersion') 
+     ? extensionHostPoolRegistration.typeHandlerVersion 
+     : '2.77'
+    autoUpgradeMinorVersion: contains(extensionHostPoolRegistration, 'autoUpgradeMinorVersion') 
+     ? extensionHostPoolRegistration.autoUpgradeMinorVersion 
+     : true
+    enableAutomaticUpgrade: contains(extensionHostPoolRegistration, 'enableAutomaticUpgrade') 
+     ? extensionHostPoolRegistration.enableAutomaticUpgrade 
+     : false
+    settings: {
+      modulesUrl: extensionHostPoolRegistration.hostPoolModulesUrl
+      configurationFunction: 'Configuration.ps1\\AddSessionHost'
+      properties: {
+        hostPoolName: extensionHostPoolRegistration.hostPoolName
+        registrationInfoToken: extensionHostPoolRegistration.registrationInfoToken
+        aadJoin: true
+      }
+    }
+    tags: extensionHostPoolRegistration.?tags ?? tags
+  }
+  dependsOn: [
+    vm_nvidiaGpuDriverWindowsExtension
+  ]
+}
+
+module vm_azureGuestConfigurationExtension 'extension/main.bicep' = if (extensionGuestConfigurationExtension.enabled) {
+  name: '${uniqueString(deployment().name, location)}-VM-GuestConfiguration'
+  params: {
+    virtualMachineName: vm.name
+    name: osType == 'Windows' ? 'AzurePolicyforWindows' : 'AzurePolicyforLinux'
+    location: location
+    publisher: 'Microsoft.GuestConfiguration'
+    type: osType == 'Windows' ? 'ConfigurationforWindows' : 'ConfigurationForLinux'
+    typeHandlerVersion: contains(extensionGuestConfigurationExtension, 'typeHandlerVersion') 
+     ? extensionGuestConfigurationExtension.typeHandlerVersion 
+     : (osType == 'Windows' ? '1.0' : '1.0')
+    autoUpgradeMinorVersion: contains(extensionGuestConfigurationExtension, 'autoUpgradeMinorVersion') 
+     ? extensionGuestConfigurationExtension.autoUpgradeMinorVersion 
+     : true
+    enableAutomaticUpgrade: contains(extensionGuestConfigurationExtension, 'enableAutomaticUpgrade') 
+     ? extensionGuestConfigurationExtension.enableAutomaticUpgrade 
+     : true
+    forceUpdateTag: contains(extensionGuestConfigurationExtension, 'forceUpdateTag') 
+     ? extensionGuestConfigurationExtension.forceUpdateTag 
+     : '1.0'
+    settings: contains(extensionGuestConfigurationExtension, 'settings') 
+     ? extensionGuestConfigurationExtension.settings 
+     : {}
+    protectedSettings: contains(extensionGuestConfigurationExtension, 'protectedSettings') 
+     ? extensionGuestConfigurationExtension.protectedSettings 
+     : {}
+    tags: extensionGuestConfigurationExtension.?tags ?? tags
+  }
+  dependsOn: [
+    vm_hostPoolRegistrationExtension
+  ]
+}
+
+resource AzureWindowsBaseline 'Microsoft.GuestConfiguration/guestConfigurationAssignments@2020-06-25' = if (!empty(guestConfiguration)) {
```

Comment:
Question: Does that need a dependency on `vm_azureGuestConfigurationExtension` as per your description of the parameter `guestConfiguration`.

Reply:
correct. fixed